modules/navier_stokes/src/actions/NSFVAction.C

```diff
@@ -2148,6 +2168,52 @@ NSFVAction::processVariables()
                      ") supplied to the NavierStokesFV action does not exist!");
 }
 
+bool
+NSFVAction::processThermalConductivity()
+{
+  checkBlockwiseConsistency<MooseFunctorName>("thermal_conductivity_blocks",
+                                              {"thermal_conductivity"});
+  bool have_scalar = false;
+  bool have_vector = false;
+
+  for (unsigned int i = 0; i < _thermal_conductivity_name.size(); ++i)
+  {
+    // First, check if the name is just a number (only in case of isotropic conduction
+    std::istringstream ss(_thermal_conductivity_name[i]);
+    Real real_value;
+    if (ss >> real_value && ss.eof())
+      have_scalar = true;
+    // Now we determine what kind of functor we are dealing with
+    else
+    {
+      if (_problem->hasFunctorWithType<ADReal>(_thermal_conductivity_name[i],
+                                               _problem->parameters().get<THREAD_ID>("_tid")))
+        have_scalar = true;
+      else
+      {
+        if (_problem->hasFunctorWithType<ADRealVectorValue>(
+                _thermal_conductivity_name[i], _problem->parameters().get<THREAD_ID>("_tid")))
+          have_vector = true;
+        else
+        {
+          if (_problem->hasFunctorWithType<Real>(_thermal_conductivity_name[i],
+                                                 _problem->parameters().get<THREAD_ID>("_tid")))
```

Comment:
@lindsayad 
what is the general use case for holding a _tid instead of retrieving it from the problem?
Do we expect the action to be threaded?

Reply:
Actions will run in serial. You can just call with a 0 tid here

Reply:
Changed thread IDs to 0!